wagtailmodelchoosers/client/components/ModelPicker.js

```diff
@@ -32,9 +32,6 @@ const propTypes = {
   onClose: PropTypes.func.isRequired,
   label: PropTypes.string.isRequired,
   endpoint: PropTypes.string.isRequired,
-  value: PropTypes.any,
-  required: PropTypes.bool.isRequired,
-  display: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
```

Comment:
Those props are indeed used by the `BaseChooser` above, not the `ModelPicker` itself. Good catch.
However, please remove `display` from the `defaultProps` above then.